src/core/plugins/db-synchronizer.ts

```diff
@@ -109,6 +114,72 @@ export default class DbSynchronizer extends CallablePlugin {
     }
   }
 
+  /**
+   This method ensures that all deployers have a complete list of active contexts.
+   Non-deployer nodes query deployers to find their own contexts. Sometimes a deployer
+   may miss some contexts. This method detects and retrieves the missing contexts from
+   other deployers and updates the own context list accordingly.
+   */
+  async deployersSyncLoop() {
+    const {monitor: {startDelay, interval}} = this.muon.configs.net.synchronizer;
+    log(`deployers sync loop start %o`, {startDelay, interval})
+
+    await timeout(Math.floor((0.5 + Math.random()) * startDelay));
+    while (true) {
+      let lastTimestamp: number = await readSetting('deployers-sync.lastTimestamp', 0);
+      log(`checking for deployer sync %o`, {lastTimestamp})
+
+      /** select random deployers */
+      const deployers: string[] = this.nodeManager
+        .filterNodes({isDeployer: true, excludeSelf: true})
+        .map(n => n.peerId);
+      const onlineDeployers: string[] = await NetworkIpc.findNOnlinePeer(deployers, 3, {timeout: 3000, return: 'peerId'});
```

Comment:
This always will select fastest nodes and could reduce the randomness. 

It is better to select a subset of deployers and then try to find 2 onlines.